src/transformers/configuration_utils.py

```diff
@@ -1078,14 +1075,29 @@ def _get_generation_defaults() -> Dict[str, Any]:
             "begin_suppress_tokens": None,
         }
 
+    def _get_non_default_generation_parameters(self) -> Dict[str, Any]:
+        """
+        Gets the non-default generation parameters on the PretrainedConfig instance
+        """
+        non_default_generation_parameters = {}
+        default_config = self.__class__()
+        for parameter_name, default_value in self._get_generation_defaults().items():
+            if hasattr(self, parameter_name):
+                # Two cases in which is okay for the model config to hold generation config parameters:
+                # 1. The parameter is set to a default generation value (from `generate`'s perspective, it's the same
+                # as if nothing is set)
+                is_default_generation_value = getattr(self, parameter_name) == default_value
+                # 2. The parameter is set as default in the model config (BC support for models like BART)
```

Comment:
This situation was not encoded before this PR (and caused tests to fail, god bless tests 🙏 )